sql/informix-sql/InformixSQLParser.g4

```diff
@@ -0,0 +1,310 @@
+    | ALL
+    | ALTER
+    | ANALYZE
+    | AND
+    | AS
+    | ASC
+    | ATTACH
+    | AUTOINCREMENT
+    | BEFORE
+    | BEGIN
+    | BETWEEN
+    | BY
+    | CASCADE
+    | CASE
+    | CAST
+    | CHECK
+    | COLLATE
+    | COLUMN
+    | COMMIT
+    | CONFLICT
+    | CONSTRAINT
+    | CREATE
+    | CROSS
+    | CURRENT_DATE
+    | CURRENT_TIME
+    | CURRENT_TIMESTAMP
+    | DATABASE
+    | DEFAULT
+    | DEFERRABLE
+    | DEFERRED
+    | DELETE
+    | DESC
+    | DETACH
+    | DISTINCT
+    | DROP
+    | EACH
+    | ELSE
+    | END
+    | ESCAPE
+    | EXCEPT
+    | EXCLUSIVE
+    | EXISTS
+    | EXPLAIN
+    | FAIL
+    | FOR
+    | FOREIGN
+    | FROM
+    | FULL
+    | GLOB
+    | GROUP
+    | HAVING
+    | IF
+    | IGNORE
+    | IMMEDIATE
+    | IN
+    | INDEX
+    | INDEXED
+    | INITIALLY
+    | INNER
+    | INSERT
+    | INSTEAD
+    | INTERSECT
+    | INTO
+    | IS
+    | ISNULL
+    | JOIN
+    | KEY
+    | LEFT
+    | LIKE
+    | LIMIT
+    | MATCH
+    | NATURAL
+    | NO
+    | NOT
+    | NOTNULL
+    | NULL
+    | OF
+    | OFFSET
+    | ON
+    | OR
+    | ORDER
+    | OUTER
+    | PLAN
+    | PRAGMA
+    | PRIMARY
+    | QUERY
+    | RAISE
+    | RECURSIVE
+    | REFERENCES
+    | REGEXP
+    | REINDEX
+    | RELEASE
+    | RENAME
+    | REPLACE
+    | RESTRICT
+    | RIGHT
+    | ROLLBACK
+    | ROW
+    | ROWS
+    | SAVEPOINT
+    | SELECT
+    | SET
+    | TABLE
+    | TEMP
+    | TEMPORARY
+    | THEN
+    | TO
+    | TRANSACTION
+    | TRIGGER
+    | UNION
+    | UNIQUE
+    | UPDATE
+    | USING
+    | VACUUM
+    | VALUES
+    | VIEW
+    | VIRTUAL
+    | WHEN
+    | WHERE
+    | WITH
+    | WITHOUT
+    | FIRST_VALUE
+    | OVER
+    | PARTITION
+    | RANGE
+    | PRECEDING
+    | UNBOUNDED
+    | CURRENT
+    | FOLLOWING
+    | CUME_DIST
+    | DENSE_RANK
+    | LAG
+    | LAST_VALUE
+    | LEAD
+    | NTH_VALUE
+    | NTILE
+    | PERCENT_RANK
+    | RANK
+    | ROW_NUMBER
+    | GENERATED
+    | ALWAYS
+    | STORED
+    | TRUE
+    | FALSE
+    | WINDOW
+    | NULLS
+    | FIRST
+    | LAST
+    | FILTER
+    | GROUPS
+    | EXCLUDE
+;
+
+name
+    : anyName
+    ;
+
+roleName
+    : anyName
+    ;
+
+functionName
+    : anyName
+    ;
+
+schemaName
+    : anyName
+    ;
+
+tableName
+    : anyName
+    ;
+
+tableOrIndexName
+    : anyName
+    ;
+
+columnName
+    : anyName
+    ;
+
+collationName
+    : anyName
+    ;
+
+foreignTable
+    : anyName
+    ;
+
+indexName
+    : anyName
+    ;
+
+triggerName
+    : anyName
+    ;
+
+viewName
+    : anyName
+    ;
+
+moduleName
+    : anyName
+    ;
+
+pragmaName
+    : anyName
+    ;
+
+savepointName
+    : anyName
+    ;
+
+tableAlias
+    : anyName
+    ;
+
+transactionName
+    : anyName
+    ;
+
+windowName
+    : anyName
+    ;
+
+alias
+    : anyName
+    ;
+
+filename
+    : anyName
+    ;
+
+baseWindowName
+    : anyName
+    ;
+
+simpleFunc
+    : anyName
+    ;
+
+aggregateFunc
+    : anyName
+    ;
+
+tableFunctionName
+    : anyName
+    ;
```

Comment:
What's the meaning of so many similar parse rules with different names?

Reply:
I thought they were useful, so I kept them.
Okay, I delete them now.